sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
Part of my objection is that `simplify` (as I think it says in the docstring) is an ill-defined function. Why aren't the existing simplifications in `checksol` handling the new case? Are steps 3-7 needed on any other case with the change you made?